frame/support/src/storage/bounded_vec.rs

```diff
@@ -176,6 +186,115 @@ impl<T, S: Get<u32>> BoundedVec<T, S> {
 		S::get() as usize
 	}
 
+	/// Forces the insertion of `s` into `self` retaining all items with index at least `index`.
+	///
+	/// If `index == 0` and `self.len() == Self::bound()`, then this is a no-op.
+	///
+	/// If `Self::bound() < index` or `self.len() < index`, then this is also a no-op.
+	///
+	/// Returns `true` if the item was inserted.
+	pub fn force_insert_keep_right(&mut self, index: usize, element: T) -> bool {
+		// Check against panics.
+		if Self::bound() < index || self.len() < index {
+			return false
+		}
+		if self.len() < Self::bound() {
+			// Cannot panic since self.len() >= index;
+			self.0.insert(index, element);
+		} else {
+			if index == 0 {
+				return false
+			}
+			self[0] = element;
```

Comment:
This here means that it will overwrite the first element when the length is equal to or greater than the bound, is that intended?

Reply:
Nevermind, the function explicitly says that it `keeps_right`, meaning whatever happens on the left is not important.